python/tank/context.py

```diff
@@ -984,7 +1107,36 @@ def context_yaml_constructor(loader, node):
 ################################################################################################
 # utility methods
 
-def _task_from_sg(tk, task_id):
+def _get_entity_type_sg_name_field(entity_type):
```

Comment:
I think we now have this code in a couple of places. Maybe this is a good time to have a util method where we can centralize the logic?